### ISSUE TITLE
NullPointerException in CertificatePinner.check()

### ISSUE THREAD

**Issue by gborland** [square/okhttp#2337]
I'm getting a few reports of this in my Crashlytics console. Using `com.squareup.okhttp3:okhttp:3.1.1`.
 
```
Fatal Exception: java.lang.NullPointerException: Attempt to invoke virtual method 'java.security.cert.X509Certificate java.security.cert.TrustAnchor.getTrustedCert()' on a null object reference
       at okhttp3.internal.tls.AndroidTrustRootIndex.findByIssuerAndSignature(AndroidTrustRootIndex.java:43)
       at okhttp3.internal.tls.CertificateChainCleaner.clean(CertificateChainCleaner.java:68)
       at okhttp3.CertificatePinner.check(CertificatePinner.java:151)
       at okhttp3.internal.io.RealConnection.connectTls(RealConnection.java:208)
       at okhttp3.internal.io.RealConnection.connectSocket(RealConnection.java:148)
       at okhttp3.internal.io.RealConnection.connect(RealConnection.java:111)
       at okhttp3.internal.http.StreamAllocation.findConnection(StreamAllocation.java:188)
       at okhttp3.internal.http.StreamAllocation.findHealthyConnection(StreamAllocation.java:127)
       at okhttp3.internal.http.StreamAllocation.newStream(StreamAllocation.java:97)
       at okhttp3.internal.http.HttpEngine.connect(HttpEngine.java:289)
       at okhttp3.internal.http.HttpEngine.sendRequest(HttpEngine.java:241)
       at okhttp3.RealCall.getResponse(RealCall.java:240)
       at okhttp3.RealCall$ApplicationInterceptorChain.proceed(RealCall.java:198)
       at okhttp3.RealCall.getResponseWithInterceptorChain(RealCall.java:160)
       at okhttp3.RealCall.access$100(RealCall.java:30)
       at okhttp3.RealCall$AsyncCall.execute(RealCall.java:127)
       at okhttp3.internal.NamedRunnable.run(NamedRunnable.java:32)
       at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
       at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
       at java.lang.Thread.run(Thread.java:818)
```

**Comment by gborland** [comment]
All of the crashes are occurring on Android 6.0.1, a mixture of rooted and non-rooted devices.

**Comment by gborland** [comment]
Closing, this is obviously a known issue. 3.1.2 didn't show up when I refreshed my dependencies, but I've got it now.

**Comment by swankjesse** [comment]
It’s an interesting bug report. It suggests that somehow the certificate isn’t trusted, but this happens after a successful TLS handshake If you have any ideas on what triggers this I’m eager to hear ’em.

**Comment by gborland** [comment]
I closed because I assumed this was one of the issues already fixed in 3.1.2. (My crash reports were coming from 3.1.1.)

I'll reopen if I get reports of it happening again with 3.1.2.
